sentry-apollo-3/src/main/java/io/sentry/apollo3/SentryApollo3HttpInterceptor.kt

```diff
@@ -0,0 +1,162 @@
+package io.sentry.apollo3
+
+import com.apollographql.apollo3.api.http.HttpRequest
+import com.apollographql.apollo3.api.http.HttpResponse
+import com.apollographql.apollo3.exception.ApolloNetworkException
+import com.apollographql.apollo3.network.http.HttpInterceptor
+import com.apollographql.apollo3.network.http.HttpInterceptorChain
+import io.sentry.Breadcrumb
+import io.sentry.Hint
+import io.sentry.HubAdapter
+import io.sentry.IHub
+import io.sentry.ISpan
+import io.sentry.SentryLevel
+import io.sentry.SpanStatus
+import io.sentry.TracingOrigins
+import io.sentry.TypeCheckHint
+import okio.Buffer
+import java.io.InputStreamReader
+
+class SentryApollo3HttpInterceptor(private val hub: IHub = HubAdapter.getInstance(), private val beforeSpan: BeforeSpanCallback? = null) :
+    HttpInterceptor {
+
+    override suspend fun intercept(
+        request: HttpRequest,
+        chain: HttpInterceptorChain
+    ): HttpResponse {
+        val activeSpan = hub.span
+        return if (activeSpan == null) {
+            chain.proceed(request)
+        } else {
+            val span = startChild(request, activeSpan)
+
+            val requestBuilder = request.newBuilder()
+
+            if (TracingOrigins.contain(hub.options.tracingOrigins, request.url)) {
+                val sentryTraceHeader = span.toSentryTrace()
+                val baggageHeader = span.toBaggageHeader()
+                requestBuilder.addHeader(sentryTraceHeader.name, sentryTraceHeader.value)
+
+                baggageHeader?.let {
+                    requestBuilder.addHeader(it.name, it.value)
+                }
+            }
+
+            val modifiedRequest = requestBuilder.build()
+
+            return try {
+                val httpResponse = chain.proceed(modifiedRequest)
+                span.status = SpanStatus.fromHttpStatusCode(httpResponse.statusCode, SpanStatus.UNKNOWN)
+                finish(span, modifiedRequest, httpResponse)
+
+                httpResponse
+            } catch (e: Exception) {
+                when (e) {
+                    is ApolloNetworkException -> span.status = SpanStatus.INTERNAL_ERROR
+                    else -> SpanStatus.UNKNOWN
```

Comment:
That looks wrong, every error would be `UNKNOWN` if not thrown by network issues, the `finally` block would be better cus there might be an error but at least we have a HTTP status code?

Reply:
Will double check but, afaik we get a httpResponse for all Status Codes including 4xx and 5xx in the HttpInterceptor